pipelined/bevy_sprite2/src/render/mod.rs

```diff
@@ -142,35 +146,75 @@ impl FromWorld for SpriteShaders {
 
 struct ExtractedSprite {
     transform: Mat4,
-    size: Vec2,
+    rect: Rect,
     handle: Handle<Image>,
+    atlas_size: Option<Vec2>,
 }
 
 pub struct ExtractedSprites {
     sprites: Vec<ExtractedSprite>,
 }
 
+pub fn extract_atlases(
+    texture_atlases: Res<Assets<TextureAtlas>>,
+    atlas_query: Query<(&TextureAtlasSprite, &GlobalTransform, &Handle<TextureAtlas>)>,
+    mut render_world: ResMut<RenderWorld>,
+) {
+    let mut extracted_sprites = Vec::new();
+    for (atlas_sprite, transform, texture_atlas_handle) in atlas_query.iter() {
+        if !texture_atlases.contains(texture_atlas_handle) {
+            continue;
+        }
+
+        if let Some(texture_atlas) = texture_atlases.get(texture_atlas_handle) {
+            let rect = texture_atlas.textures[atlas_sprite.index as usize];
+            extracted_sprites.push(ExtractedSprite {
+                atlas_size: Some(texture_atlas.size),
+                transform: transform.compute_matrix(),
+                rect: rect.clone(),
+                handle: texture_atlas.texture.clone_weak(),
+            });
+        }
+    }
+
+    if let Some(mut extracted_sprites_res) = render_world.get_resource_mut::<ExtractedSprites>() {
+        extracted_sprites_res.sprites.extend(extracted_sprites);
+    } else {
+        render_world.insert_resource(ExtractedSprites {
+            sprites: extracted_sprites,
+        });
+    }
+}
+
 pub fn extract_sprites(
-    mut commands: Commands,
     images: Res<Assets<Image>>,
-    query: Query<(&Sprite, &GlobalTransform, &Handle<Image>)>,
+    sprite_query: Query<(&Sprite, &GlobalTransform, &Handle<Image>)>,
+    mut render_world: ResMut<RenderWorld>,
 ) {
     let mut extracted_sprites = Vec::new();
-    for (sprite, transform, handle) in query.iter() {
+    for (sprite, transform, handle) in sprite_query.iter() {
         if !images.contains(handle) {
             continue;
         }
 
         extracted_sprites.push(ExtractedSprite {
+            atlas_size: None,
             transform: transform.compute_matrix(),
-            size: sprite.size,
+            rect: Rect {
+                min: Vec2::ZERO,
+                max: sprite.size,
```

Comment:
It's worth pointing out that this actually changes existing behavior for `sprite.size` in the fact that it now indicates a pixel texel size instead of a sprite dimension size. Instead it might be better to make `Rect` optional here?